src/compiler/transformers/es2015.ts

```diff
@@ -1024,8 +1024,11 @@ namespace ts {
                 }
             }
 
-            // Return the result if we have an immediate super() call on the last statement.
-            if (superCallExpression && statementOffset === ctorStatements.length - 1) {
+            // Return the result if we have an immediate super() call on the last statement,
+            // but only if the constructor itself doesn't use 'this' elsewhere.
+            if (superCallExpression
+            && statementOffset === ctorStatements.length - 1
+            && !(ctor.transformFlags & (TransformFlags.ContainsLexicalThis | TransformFlags.ContainsCapturedLexicalThis))) {

```

Comment:
No, it's only used during aggregation to determine whether to promote `ContainsLexicalThis` in a computed property name so that it doesn't get stopped at a method or accessor.